src/messaging/data/mod.rs

```diff
@@ -144,28 +150,65 @@ impl QueryResponse {
                 Err(error) => matches!(*error, ErrorMessage::DataNotFound(_)),
             },
-            GetRegister(result) => match result {
+            GetRegister((result, _xorname)) => match result {
                 Ok(_) => false,
                 Err(error) => matches!(*error, ErrorMessage::DataNotFound(_)),
             },
-            GetRegisterOwner(result) => match result {
+            GetRegisterOwner((result, _xorname)) => match result {
                 Ok(_) => false,
                 Err(error) => matches!(*error, ErrorMessage::DataNotFound(_)),
             },
-            ReadRegister(result) => match result {
+            ReadRegister((result, _xorname)) => match result {
                 Ok(_) => false,
                 Err(error) => matches!(*error, ErrorMessage::DataNotFound(_)),
             },
-            GetRegisterPolicy(result) => match result {
+            GetRegisterPolicy((result, _xorname)) => match result {
                 Ok(_) => false,
                 Err(error) => matches!(*error, ErrorMessage::DataNotFound(_)),
             },
-            GetRegisterUserPermissions(result) => match result {
+            GetRegisterUserPermissions((result, _xorname)) => match result {
                 Ok(_) => false,
                 Err(error) => matches!(*error, ErrorMessage::DataNotFound(_)),
             },
         }
     }
+
+    /// Retrieves the operation identifier for this response, use in tracking node liveness
+    /// and responses at clients.
+    /// Right now returning result to fail for anything non-chunk, as that's all we're tracking from other nodes here just now.
+    pub fn operation_id(&self) -> Result<OperationId> {
+        use QueryResponse::*;
+        let mut hasher = DefaultHasher::new();
+
+        // TODO: Operation Id should eventually encompass _who_ the op is for.
+        match self {
+            GetChunk(result) => match result {
+                Ok(chunk) => {
+                    chunk.address().hash(&mut hasher);
+
+                    Ok(hasher.finish())
+                }
+                Err(error) => match error {
+                    ErrorMessage::DataNotFound(address) => match address {
+                        DataAddress::Chunk(address) => {
+                            address.hash(&mut hasher);
+
+                            Ok(hasher.finish())
+                        }
+                        _ => Err(Error::NoOperationIdForElderData),
+                    },
+                    _ => Err(Error::InvalidQueryResponseErrorForOperationId),
+                },
+            },
+
+            GetRegister((_result, operation_id)) => Ok(*operation_id),
+            GetRegisterOwner((_result, operation_id)) => Ok(*operation_id),
+            ReadRegister((_result, operation_id)) => Ok(*operation_id),
+            GetRegisterPolicy((_result, operation_id)) => Ok(*operation_id),
+            GetRegisterUserPermissions((_result, operation_id)) => Ok(*operation_id),
```

Comment:
Nitpick: these could be combined:

```suggestion
            GetRegister((_result, operation_id))
            | GetRegisterOwner((_result, operation_id))
            | ReadRegister((_result, operation_id))
            | GetRegisterPolicy((_result, operation_id))
            | GetRegisterUserPermissions((_result, operation_id)) => Ok(*operation_id),
```